lib/rspec/core/configuration.rb

```diff
@@ -1613,6 +1613,9 @@ def extract_location(path)
           captures = match.captures
           path, lines = captures[0], captures[1][1..-1].split(":").map { |n| n.to_i }
           filter_manager.add_location path, lines
+        else
+          path, scoped_ids = path.split(/[\[\]]/)
```

Comment:
I personally find that to be less readable than what's there now, and it's almost certainly going to perform slower (it has the extra `Regexp.escape` method call plus the string interpolation, plus the extra objects created by `"[]"` and `Regexp.escape`...).